cdbot/cogs/admin.py

```diff
@@ -89,6 +91,50 @@ async def on_member_join(self, member: Member):
             # assign placeholder nickname
             await member.edit(nick=PLACEHOLDER_NICKNAME)
 
+    @command()
+    @has_role(ROOT_ROLE_ID)
+    async def raidprotect(
+        self,
+        ctx: Context
+    ):
+        """
+        Allows an admin user to lock down the server in case of a raid.
+        This command toggles invite link generation for @everyone and
+        revokes all existing invite links.
+        """
+        everyone = ctx.channel.guild.default_role
+        perms = everyone.permissions
+        logs_channel = self.bot.get_channel(LOGGING_CHANNEL_ID)
+
+        if perms.create_instant_invite:
+            perms.update(create_instant_invite=False)
+            raidon = 1
+        else:
+            perms.update(create_instant_invite=True)
+            raidon = 0
+
+        msgpart = ("OFF", "ON")
+        msg = f"Toggled raid protection {msgpart[raidon]}."
+        await everyone.edit(reason=msg, permissions=perms)
+        await ctx.send(msg, delete_after=10)
+        embed = Embed(
+            color=Colour.blue(),
+            title=f"Raid Protection {msgpart[raidon]}."
+        )
+        if raidon:
+            embed.description = ("Raid protection now ON - All invite "
+                                 "links were deleted and members may not "
+                                 "create new ones")
+        else:
+            embed.description = ("Raid protection now OFF - Members can now "
+                                 "create new invite links")
+        embed.set_author(name=ctx.author.name, icon_url=ctx.author.avatar_url)
+        await logs_channel.send(embed=embed)
+
+        if raidon:
```

Comment:
Could this be placed below the first `if raidon` clause below line 127 instead? So it preps the embed and revokes the roles all under one if statement?

Reply:
Yep, I can do that now